pkg/sink/sink_test.go

```diff
@@ -179,687 +179,758 @@ func checkSinkResponse(t *testing.T, resp *http.Response, elName string) {
+		webhookInterceptor: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// Copy over all headers
+			for k := range r.Header {
+				for _, v := range r.Header[k] {
+					w.Header().Add(k, v)
+				}
+			}
+			// Read the Name header
+			var name string
+			if nameValue, ok := r.Header["Name"]; ok {
+				name = nameValue[0]
+			}
+			// Write the name to the body
+			body := fmt.Sprintf(`{"name": "%s"}`, name)
+			_, _ = w.Write([]byte(body))
+		}),
+		eventBody: eventBody,
+		want: []pipelinev1.TaskRun{{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "tekton.dev/v1beta1",
+				Kind:       "TaskRun",
 			},
-		},
-		Spec: pipelinev1alpha1.PipelineResourceSpec{
-			Type: pipelinev1alpha1.PipelineResourceTypeGit,
-			Params: []pipelinev1alpha1.ResourceParam{
-				{Name: "url", Value: "testurl"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "name-from-webhook",
+				Namespace: namespace,
+				Labels: map[string]string{
+					"app":                                  "triggers",
+					"type":                                 "bar",
+					"triggers.tekton.dev/eventlistener":    eventListenerName,
+					"triggers.tekton.dev/trigger":          "git-clone-trigger",
+					"triggers.tekton.dev/triggers-eventid": "12345",
+				},
 			},
-		},
+			Spec: pipelinev1.TaskRunSpec{
+				Params: []pipelinev1.Param{{
+					Name:  "url",
+					Value: pipelinev1.ArrayOrString{Type: pipelinev1.ParamTypeString, StringVal: "https://github.com/tektoncd/triggers"},
+				}, {
+					Name:  "git-revision",
+					Value: pipelinev1.ArrayOrString{Type: pipelinev1.ParamTypeString, StringVal: "master"},
+				}},
+				TaskRef: &pipelinev1.TaskRef{Name: "git-clone"},
+			},
+		}},
 	}}
-	gotPrs := getCreatedPipelineResources(t, dynamicClient.Actions())
-	if diff := cmp.Diff(wantResource, gotPrs); diff != "" {
-		t.Errorf("Created resources mismatch (-want + got): %s", diff)
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// TODO: Do we ever support multiple eventListeners? Maybe change test.Resources to only accept one?
+			elName := tc.resources.EventListeners[0].Name
+			sink, dynamicClient := getSinkAssets(t, tc.resources, elName, tc.webhookInterceptor)
+			ts := httptest.NewServer(http.HandlerFunc(sink.HandleEvent))
+			defer ts.Close()
+			req, err := http.NewRequest("POST", ts.URL, bytes.NewReader(tc.eventBody))
+			if err != nil {
+				t.Fatalf("error creating request: %s", err)
+			}
+			if tc.headers != nil {
+				req.Header = http.Header(tc.headers)
+			}
+			req.Header.Add("Content-Type", "application/json")
+			resp, err := http.DefaultClient.Do(req)
+			if err != nil {
+				t.Fatalf("error sending request: %s", err)
+			}
+			checkSinkResponse(t, resp, elName)
+			// Check right resources were created.
+			got := toTaskRun(t, dynamicClient.Actions())
+			// Sort actions (we do not know what order they executed in)
+			if diff := cmp.Diff(tc.want, got, cmpopts.SortSlices(func(x, y pipelinev1.TaskRun) bool {
```

Comment:
Could this be split into two lines? Otherwise the comment is not really true as we sort and compare in a single line. It would make the code more readable / easier to follow IMO.

Reply:
sure!

Reply:
Updated the comment -- we don't technically sort the taskRuns ourselves, `cmp.Diff` has an option to sortSlices, we just pass it a function that it can use to compare the TaskRuns when sorting.